pandas/core/arrays/datetimelike.py

```diff
@@ -1079,6 +1397,44 @@ def _evaluate_compare(self, other, op):
         result[mask] = filler
         return result
 
+    # --------------------------------------------------------------
+    # Reductions
+
+    def _reduce(self, name, skipna=True, **kwargs):
+        op = getattr(self, name, None)
+        if op:
+            return op(skipna=skipna)
+        else:
+            return super(DatetimeLikeArrayMixin, self)._reduce(
+                name, skipna, **kwargs
+            )
+
+    def min(self, skipna=True):
+        result = nanops.nanmin(self.asi8, skipna=skipna, mask=self.isna())
+        if isna(result):
+            # Period._from_ordinal does not handle np.nan gracefully
+            return NaT
+        return self._box_func(result)
+
+    def max(self, skipna=True):
+        # TODO: skipna is broken with max.
+        # See https://github.com/pandas-dev/pandas/issues/24265
+        mask = self.isna()
+        if skipna:
+            values = self[~mask].asi8
+        elif mask.any():
+            return NaT
+        else:
+            values = self.asi8
+
+        if not len(values):
+            # short-circut for empty max / min
+            return NaT
+        # Do not pass mask, since it's maybe not the right shape.
```

Comment:
huh? why would this be true?

Reply:
If `skipna` is true and there are missing values. `mask` will be longer that `values`.